### DISCUSSION TITLE
Debugging createResolvers resolve function

### DISCUSSION THREAD

**Discussion by issc29** [gatsbyjs/gatsby#35977] [Help] [answered]
I am using the `createResolvers` in gatsby-node but need to debug something in the the `resolve` function. Is there a good way to debug this function? I tried using the VS Code debugger but it doesnt seem to stop at the breakpoints and console.log statements don't seem to make it to the console.

Example:

```
exports.createResolvers = ({createResolvers}) => {
  createResolvers({
    track: {
      allTags: {
        type: 'String',
        resolve(source, args, context, info) {
          console.log("TEST CONSOLE")
          console.log(source)

          return "TEST"
        }
      }
    }
  })
}
```

**Comment by issc29** [comment, marked as answer]
I figured it out, needed to actually run a graphql query for the resolvers to run

**Reply by fmaiabatista** [reply]
Can you elaborate?

I'm running into the same problem and I cannot make it work by following the example code from the documentation: https://www.gatsbyjs.com/docs/reference/graphql-data-layer/schema-customization/#accessing-gatsbys-data-store-from-field-resolvers 🤔

**Reply by fmaiabatista** [reply]
Oh wow, after a while I figured out the whole process.

I documented it here so future visitors can refer to it:

![image](https://user-images.githubusercontent.com/1574224/220997636-b5ba2237-00fb-4fd5-8ab8-fa0a4931fea3.png)

Note: when I first read `actually run a graphql query for the resolvers to run` I thought I had to call the [`.runQuery` or similar method](https://www.gatsbyjs.com/docs/reference/release-notes/migrating-from-v3-to-v4/#nodemodelrunquery-is-deprecated) lolz. Only now after figuring it out, I read it differently... You have to actually execute a query via Gatsby's GraphiQL interface (in your browser) to see the log in your terminal! 👌
